internal/armada/configuration/types.go

```diff
@@ -88,6 +88,28 @@ type PulsarConfig struct {
 	BackoffTime time.Duration
 }
 
+// DatabaseConfig represents the configuration of the database connection.
+type DatabaseConfig interface {
```

Comment:
I don't think we allow functions in our yaml configurations so this may be a bit weird..  

Looking at our configuration we only allow things you can pass into raw yaml so not sure what this interface would be used for.

Reply:
agree, can we just use the struct properties without the getter functions?

Reply:
yes makes sense, a struct is more convenient, I'll update the PR accordingly. thanks!